src/attribution/onINP.ts

```diff
@@ -202,39 +202,45 @@ export const onINP = (
   };
 
   const cleanupEntries = () => {
-    // Keep all render times that are part of a pending INP candidate or
-    // that occurred within the 50 most recently-dispatched groups of events.
-    const longestInteractionGroups =
+    // Create a set of entries groups that are part of the longest
+    // interactions (for faster lookup below).
+    const longestInteractionGroups = new Set(
       interactionManager._longestInteractionList.map((i) => {
         return entryToEntriesGroupMap.get(i.entries[0]);
-      });
-    const minIndex = pendingEntriesGroups.length - MAX_PREVIOUS_FRAMES;
-    pendingEntriesGroups = pendingEntriesGroups.filter((group, index) => {
-      if (index >= minIndex) return true;
-      return longestInteractionGroups.includes(group);
+      }),
+    );
+
+    // Clean up the `pendingEntriesGroups` list so it doesn't grow endlessly.
+    // Keep any groups that:
+    // 1) Correspond to one of the current longest interactions, OR
+    // 2) Any group that's part of the most recent set of frames (based on
+    //    `MAX_PENDING_FRAMES`).
+    pendingEntriesGroups = pendingEntriesGroups.filter((group, i) => {
+      return (
+        longestInteractionGroups.has(group) ||
+        i >= pendingEntriesGroups.length - MAX_PENDING_FRAMES
+      );
     });
 
-    // Keep all pending LoAF entries that either:
-    // 1) intersect with entries in the newly cleaned up `pendingEntriesGroups`
-    // 2) occur after the most recently-processed event entry (for up to MAX_PREVIOUS_FRAMES)
-    const loafsToKeep: Set<PerformanceLongAnimationFrameTiming> = new Set();
```

Comment:
Renaming `loafsToKeep` to `intersectionLoAFs` because these aren't the only LoAFs that are being kept.